lib/elixir/lib/map_set.ex

```diff
@@ -84,13 +94,40 @@ defmodule MapSet do
 
   """
   @spec difference(t, t) :: t
-  def difference(%MapSet{map: map1}, %MapSet{map: map2}) do
-    map = :maps.fold(fn value, _, acc ->
-      Map.delete(acc, value)
-    end, map1, map2)
+  # If the first set is less than twice the size of the second map,
+  # it is fastest to re-accumulate items in the first set that are not
+  # present in the second set.
+  def difference(%MapSet{map: map1}, %MapSet{map: map2})
+    when map_size(map1) < map_size(map2) * 2 do
```

Comment:
It think `when` should be on same level as `def`.
There are currently only two styles can be seen:

* same indentation level as for `def`
* 2 extra levels (4 spaces) from `def`